app/models.py

```diff
@@ -437,6 +473,21 @@ class UserEvent(Event):
     def from_user(cls, user, **kwargs):
         return cls(user_id=user.id, deployment=user.deployment, **kwargs)
 
+class SimpleUserEvent(UserEvent):
+    '''
+    Any kind of user event that doesn't have any extra data associated
+    with it.
+    '''
+
+    # A user completed the registration process and joined the network.
+    SUBTYPE_USER_JOINED = 1
+
+    subtype = Column(types.Integer)
+
+    __mapper_args__ = {
+        'polymorphic_identity': 'simple_user_event'
+    }
```

Comment:
I think it'd be better to do this by having subclasses of `SimpleUserEvent` for each user event type, where the `polymorphic_identity` then gets set to what sort of event it is (for example, `user_joined`).  That way, we can do queries like:

```
UserJoinedEvent.query.filter_by(user_id=<user_id>).exists()
```

To determine whether the user has an event associated without directly querying the subtype.